qa/tasks/cephadm.py

```diff
@@ -2179,6 +2179,9 @@ def task(ctx, config):
     containers = config.get('containers', {})
     container_image_name = containers.get('image', container_image_name)
 
+    if 'force_image' in config:
```

Comment:
I think this does the same thing as config.image.  Here's a bit of doc I wrote that matches my experience with the existing code:

+    Set cephadm.image to supply the whole image name, including version.
+    Set cephadm.containers.image to an image tag without version, and the task
+     will use the toplevel config items sha1:, flavor:, and branch: to
+     create either :sha1, :sha1-flavor:, or :branch (default main) as a version
+

Reply:
Interesting, thanks.  I mentioned to @adk3798 earlier today I had trouble testing the 1st patch without these hacks but I could have messed something up. If @adk3798 agrees that this is not needed I can drop this patch from the PR and then I will retest at my leisure later and if I find I still need this for some reason I'll try posting it again.

I find the code that implements all this derivation of the container name incredibly confusing to the point that I tried and failed to refactor it once before! Maybe the confusing code prevents me from understanding the intended workflow ;-)

Reply:
the code certainly could use some inline comments too as well as the above.